### PR TITLE
Amit/roger 968 Created common role of permissions generation for nginx and marathon-server roles

### PR DESCRIPTION
**Pull request by amitbose327**
1. Moved our temp dir creation, temp dir cleanup handler, creating user-permissions.json from google spreadsheets, creating nginx basic auth file (.htpasswd) and copying the permission files to servers all in a common role (generate-permissions-file).
2. This common role is depended by nginx and marathon-server and run when enable_nginx_auth and marathon_auth_enable are true respectively.
3. Using virtualenv to install python packages to locally generate the permissions file from google spreadsheets.
4. Three options are supported to copy the permissions file for either role. (From googlespreadsheets, local file passed in, or default template file)
5. Renamed ansible secrets variables and places they are used. Pushed updated secrets file.

@ankanm and @vmahedia Please review.

### CODE REVIEW

**Comment by vmahedia** [comment]
LGTM

File: ansible/roles/generate-permission-files/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    copy: src="{{ creds_dir }}/permissions.json" dest="{{ marathon_auth_plugin_location }}/user-permissions.json" owner=root group=root mode=0644
+    become: yes
+    tags:
+      - configuration
+      - plugin
+      - permissions
+      - permissions-sync
+      - permissions-gen
+
+  - name: Copy permissions file from local src when marathon_auth_permissions_local_path is defined
```
Comment (ankanm) [inline ansible/roles/generate-permission-files/tasks/main.yml]: Isn't `marathon_auth*` vars now `auth*`?

Comment (amitbose327) [inline ansible/roles/generate-permission-files/tasks/main.yml]: Not all marathon_auth vars. Only the ones that are common for both marathon and nginx have changed.

**Comment by ankanm** [comment]
LGTM

**Pull request merged**